### PR TITLE
PIMOB-347: Add NetworkError tests

### PR DESCRIPTION
**Pull request by daven-gomes-cko**
## Proposed changes

Fix logic in `NetworkError` where it would always return `.unknown` in the `init` method.
Add unit tests for `NetworkError`.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [X] Bugfix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [X] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Tests/NetworkErrorTests.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+//  Created by Daven.Gomes on 02/12/2020.
+//  Copyright © 2020 Checkout. All rights reserved.
+//
+
+import XCTest
+@testable import FramesIos
+
+final class NetworkErrorTests: XCTestCase {
+
+    func test_init_allValuesExist_checkoutErrorReturned() {
```
Comment (deepesh-vasthimal-cko) [inline Tests/NetworkErrorTests.swift:14]: 👍

File: Tests/NetworkErrorTests.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            XCTAssertEqual(errorType, "stubErrorType")
+            XCTAssertEqual(errorCodes, ["stubErrorCode1",
+                                        "stubErrorCode2"])
+
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func test_init_requestIDNil_returnUnknown() {
```
Comment (deepesh-vasthimal-cko) [inline Tests/NetworkErrorTests.swift:42]: if this test is returning  Unknown should we not do a XCTAssertEqual for unknown ?

File: Tests/NetworkErrorTests.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            guard case .unknown = result else {
+                return XCTFail("Unexpected NetworkError type.")
+            }
+
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func test_init_errorTypeNil_returnUnknown() {
```
Comment (deepesh-vasthimal-cko) [inline Tests/NetworkErrorTests.swift:65]: if this test is returning  Unknown should we also not do a XCTAssertEqual for unknown ?

File: Tests/NetworkErrorTests.swift [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            guard case .unknown = result else {
+                return XCTFail("Unexpected NetworkError type.")
+            }
+
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func test_init_errorTypeErrorCodes_returnUnknown() {
```
Comment (deepesh-vasthimal-cko) [inline Tests/NetworkErrorTests.swift:88]: if this test is returning Unknown should we also do a XCTAssertEqual for unknown ?

File: Tests/NetworkErrorTests.swift (hunk 2 above)
Comment (daven-gomes-cko) [reply Tests/NetworkErrorTests.swift:42]: we'd have to make `NetworkError` conform to `Equatable`, do we want to do that? I'm a bit unsure about adding protocol conformance only for unit tests 🤔

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Pull request merged**